test/drop_test/peer_resolver.py

```diff
@@ -4,31 +4,9 @@
 import time
 
 
-class Peer:
-    def __init__(self, name: str, scenario: str):
-        self._name = name
-        self._scenario = scenario
-
-    @classmethod
-    def from_hostname(cls, hostname: str) -> "Peer":
-        print(f"Creating peer from hostname {hostname}", flush=True)
-        (name, scenario) = hostname.split("-", maxsplit=1)
-        return cls(name, scenario)
-
-    def get_hostname(self) -> str:
-        return f"{self._name}-{self._scenario}"
-
-    def get_name(self) -> str:
-        return self._name
-
-    def get_scenario(self) -> str:
-        return self._scenario
-
-
 class PEERResolver:
     def __init__(self):
         self._peer_mappings = {}
-        self._cache = {}
```

Comment:
Could you elaborate on why was the caching removed?

Reply:
Sure! So I noticed that caching is unnecessary as we can directly probe OS for reverse DNS lookup. Caching works fine as well however it makes it strict that a peer must be first queried with its name. I just refactored code and it seemed to work while being simpler to manage